core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1135,6 +1137,22 @@ private[spark] class DAGScheduler(
     }
   }
 
+  /**
+   * PythonRunner needs to know what the pyspark memory setting is for the profile being run.
+   *  Pass it in the local properties of the task if it's set for the stage profile.
+   */
+  private def addPysparkMemToProperties(stage: Stage, properties: Properties): Unit = {
+    val pysparkMem = if (stage.resourceProfileId == DEFAULT_RESOURCE_PROFILE_ID) {
+      logDebug("Using the default pyspark executor memory")
+      sc.conf.get(PYSPARK_EXECUTOR_MEMORY)
+    } else {
+      val rp = sc.resourceProfileManager.resourceProfileFromId(stage.resourceProfileId)
+      logDebug(s"Using profile ${stage.resourceProfileId} pyspark executor memory")
+      rp.getPysparkMemory
+    }
```

Comment:
Can't default resource profile gets pySpark mem from `getPysparkMemory`?

Reply:
no, the ResourceProfiles doesn't have the SparkConf. I could make a utility function that takes the conf as a parameter and move it into there. I actually thought about that but it requires a bunch of parameters or still doing some logic here because it either uses the profile id, or it uses the profiles itself. This just seemed a bit cleaner here, but I can move it if you have a strong opinion

Reply:
Can't we initialize pySpark memory within `ResourceProfile.getOrCreateDefaultProfile(conf)`?

Reply:
yes it is